DateTimeCalendar/Date.java

```diff
@@ -0,0 +1,419 @@
+                payload += Date.Month.values()[i].days + 1;
+            } else {
+                payload += Date.Month.values()[i].days;
+            }
+        }
+
+        return payload;
+    }
+
+    public static boolean isLeapYear(int year){
+        return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+    }
+
+    public enum Month {
+        JAN("Jan", "January", 31),
+        FEB("Feb", "Febuary", 28),
+        MAR("Mar", "March", 31),
+        APR("Apr", "April", 30),
+        MAY("May", "May", 31),
+        JUN("Jun", "June", 30),
+        JUL("Jul", "July", 31),
+        AUG("Aug", "August", 31),
+        SEP("Sep", "September", 30),
+        OCT("Oct", "October", 31),
+        NOV("Nov", "November", 30),
+        DEC("Dec", "December", 31),
+        ;
+
+        public final String shortName;
+        public final String longName;
+        public final int days;
+
+        private Month(final String shortName, final String longName, final int days){
+            this.shortName = shortName;
+            this.longName = longName;
+            this.days = days;
+        }
+
+        //CAL~Already exists under enum class[compareTO] made class to see how we would implement if required
+        // public int diff(Month input){
+        //     return this.ordinal() - input.ordinal();
+
+        //     // if (this.ordinal() == input.ordinal()) {
+        //     //     return 0;
+        //     // } else if (this.ordinal() > input.ordinal()){
+        //     //     return 1;
+        //     // } else {
+        //     //     return -1;
+        //     // }
+        // }
+    }
+
+    private int dayspan;
+    private int year;
+    private Month month;
+    private int day;
+
+// Constructors
+    private Date(){
+        this.dayspan = 0;
+    }
+
+    public Date(int year, Month month, int day){
+        setYear(year);
+        setDay(day);
+        setMonth(month);
+    }
+
+    public Date(Month month){
+        setMonth(month);
+    }
+
+    public Date(int input){
+        setDayspan(input);
+    }
+// End Constructors
+
+// Methods
+    public Date subtract(int years, int months, int days){
+        //what if any number goes below zero?
+        /*
+        Order of opperation
+        need to remove years -> months -> days
+        - No negatives for input
+        - should I make multiple objects and do years/months/days one by one or just have the extra logic
+        - are the days greater than 365?
+        */
+        final int monthsInYear = 12;
+        int modifiedYear = this.year;
+        int modifiedMonth = this.month.ordinal() + 1;
+        int modifiedDay = this.day;
+
+        if (years >= 0 && months >= 0 && days >= 0) {
+            //years
+            //remove full years, from days that are greater than 366 and months greater than 12
+            while (years > 0 || months > 12 || days > 366) {
+                modifiedYear -= years;
+                years = 0;
```

Comment:
Technically you can take these two lines out of the `while` to right before it; that way they're executed only once

Reply:
Done!